client.go

```diff
@@ -186,6 +201,15 @@ func (p *s3Pointer) MarshalJSON() ([]byte, error) {
 	return []byte(fmt.Sprintf(`["%s",{"s3BucketName":"%s","s3Key":"%s"}]`, p.class, p.S3BucketName, p.S3Key)), nil
 }
 
+// generateS3Key returns the object key, attaches the prefix if it exists (prefix/filename)
+func generateS3Key(prefix string, filename string) string {
```

Comment:
Could this function be transitioned to be a private method on the `Client` struct? I can't see a use case where `prefix` will not be just set to `c.objectPrefix`.

Reply:
@jwcorle thanks for the feedback. I have pushed a commit that I think address your request.

Reply:
Yep, thanks!